optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -625,4 +643,231 @@ private boolean shouldIgnoreMember(ClassInfo declaringClass) {
         return declaringClass.name().toString().startsWith(SolutionDescriptor.class.getName());
     }
 
+    private void registerCustomClassesFromSolverConfig(SolverConfig solverConfig, Set<Class<?>> reflectiveClassSet) {
+        ScoreDirectorFactoryConfig scoreDirectorFactoryConfig = solverConfig.getScoreDirectorFactoryConfig();
+        while (scoreDirectorFactoryConfig != null) {
+            registerClassForReflection(scoreDirectorFactoryConfig.getEasyScoreCalculatorClass(), reflectiveClassSet);
+            registerClassForReflection(scoreDirectorFactoryConfig.getConstraintProviderClass(), reflectiveClassSet);
+            registerClassForReflection(scoreDirectorFactoryConfig.getIncrementalScoreCalculatorClass(), reflectiveClassSet);
+            scoreDirectorFactoryConfig = scoreDirectorFactoryConfig.getAssertionScoreDirectorFactory();
+        }
+
+        if (solverConfig.getPhaseConfigList() != null) {
+            for (PhaseConfig<?> phaseConfig : solverConfig.getPhaseConfigList()) {
+                registerPhaseConfigForReflection(phaseConfig, reflectiveClassSet);
+            }
+        }
+
+        registerClassForReflection(solverConfig.getRandomFactoryClass(), reflectiveClassSet);
+        registerTerminationConfig(solverConfig.getTerminationConfig(), reflectiveClassSet);
+    }
+
+    private void registerClassForReflection(Class<?> clazz, Set<Class<?>> reflectiveClassSet) {
+        if (clazz != null) {
+            reflectiveClassSet.add(clazz);
+        }
+    }
+
+    private void registerPhaseConfigForReflection(PhaseConfig<?> phaseConfig, Set<Class<?>> reflectiveClassSet) {
+        if (phaseConfig == null) {
```

Comment:
getPhaseConfigList can not contain null elements

Reply:
But I doubt that that validation triggers before this code is run, so this is valid.